yb-voyager/src/tgtdb/attr_name_registry.go

```diff
@@ -6,51 +6,67 @@ import (
 	"sync"
 
 	log "github.com/sirupsen/logrus"
-	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
 	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils/sqlname"
 	"golang.org/x/exp/slices"
 )
 
 type AttributeNameRegistry struct {
-	dbType    string
-	tdb       TargetDB
-	tconf     *TargetConf
-	attrNames *utils.StructMap[sqlname.NameTuple, []string]
-	mu        sync.Mutex
+	dbType                  string
+	tdb                     TargetDB
+	tconf                   *TargetConf
+	attrNames               sync.Map
+	bestMatchingColumnCache sync.Map
 }
 
 func NewAttributeNameRegistry(tdb TargetDB, tconf *TargetConf) *AttributeNameRegistry {
 	return &AttributeNameRegistry{
-		dbType:    tconf.TargetDBType,
-		tdb:       tdb,
-		tconf:     tconf,
-		attrNames: utils.NewStructMap[sqlname.NameTuple, []string](),
+		dbType:                  tconf.TargetDBType,
+		tdb:                     tdb,
+		tconf:                   tconf,
+		attrNames:               sync.Map{}, //sqlname.NameTuple.ForKey() -> []string
+		bestMatchingColumnCache: sync.Map{}, //sqlname.NameTuple.ForKey() -> sync.Map{}
 	}
 }
 
 func (reg *AttributeNameRegistry) QuoteAttributeName(tableNameTup sqlname.NameTuple, columnName string) (string, error) {
 	var err error
-	targetColumns, ok := reg.attrNames.Get(tableNameTup)
+	anyArr, ok := reg.attrNames.Load(tableNameTup.ForKey())
+	var targetColumns []string
+	if anyArr != nil {
+		targetColumns = anyArr.([]string)
+	}
 	if !ok {
-		reg.mu.Lock()
-		// try again in case it's now available
-		targetColumns, ok = reg.attrNames.Get(tableNameTup)
-		if !ok {
-			targetColumns, err = reg.tdb.GetListOfTableAttributes(tableNameTup)
-			log.Infof("columns of table %s in target db: %v", tableNameTup, targetColumns)
-			if err != nil {
-				return "", fmt.Errorf("get list of table attributes: %w", err)
-			}
-			reg.attrNames.Put(tableNameTup, targetColumns)
+		targetColumns, err = reg.tdb.GetListOfTableAttributes(tableNameTup)
+		log.Infof("columns of table %s in target db: %v", tableNameTup, targetColumns)
+		if err != nil {
+			return "", fmt.Errorf("get list of table attributes: %w", err)
 		}
+		reg.attrNames.Store(tableNameTup.ForKey(), targetColumns)
+	}
+	return reg.withCacheFindBestMatchingColumnName(columnName, targetColumns, tableNameTup)
```

Comment:
You can lookup from cache at the start of this function (without even having to deal with getting list of table attributes, since you don't really need that for cache lookup). Similar to what amit did in https://github.com/yugabyte/yb-voyager/pull/1433/files